### PR TITLE
[13.0][FIX] purchase_stock lines date propagation

### PR DESCRIPTION
**Pull request by chienandalu**
This fix landed in [saas-13.4](https://github.com/odoo/odoo/tree/saas-13.4) and it was later discarded as in [14.0](https://github.com/odoo/odoo/tree/14.0) all the propagate_date stuff was discarded. Anyway, the bug still exists in [13.0](https://github.com/odoo/odoo/tree/13.0).

From the rationale of the commit:

Before this commit, when we manually create a PO line, the propagate_date will always be `False`. So the change on scheduled date will never pass to corresponding move.

After this commit, when create a PO line, if no propagate_date for it, we find if the product has a route with a buy rule. If yes, use the setting on its first buy rule, if no, set the default to be true.

cc @Tecnativa TT37567

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by Whenrow** [comment]
robodoo r+
see https://github.com/odoo/odoo/pull/55155 to get full context
@chienandalu Thank you

**Comment by Whenrow** [comment]
robodoo rebase-ff

**Comment by chienandalu** [comment]
@Whenrow It seems didn't make it into the main branch. Can you check why?

File: addons/purchase_stock/models/purchase.py [hunk 1]
```diff
@@ -274,12 +274,29 @@ def _compute_qty_received(self):
...
-        if line.order_id.state == 'purchase':
-            line._create_or_update_picking()
-        return line
+    @api.model_create_multi
+    def create(self, vals_list):
+        lines = super(PurchaseOrderLine, self).create(vals_list)
+
+        # when no "propagate_date", find if the product has a route with a buy rule,
+        # if yes, use the setting on the rule, if no, set "True" as default.
+        for line, vals in zip(lines, vals_list):
```
Comment (cesar-tecnativa) [inline addons/purchase_stock/models/purchase.py:283]: `if 'propagate_date' not in vals`  before for loop to prevent going inside?

**Review by cesar-tecnativa — CHANGES_REQUESTED** [review]

**Comment by Whenrow** [comment]
robodoo retry

**Comment by pedrobaeza** [comment]
Thanks!

**Pull request closed without merging**